### PR TITLE
Feature/members only

### PR DESCRIPTION
**Pull request by Redvanisation**
### Members only app:

- Please use the console to create new users.
- Logged in users can publish new articles and view who wrote the other articles.
- If you logged out you can see the articles list without the writer's name.
- You need to login first to access the writing a new article page, so if you aren't logged in you will be redirected to the login page.

### CODE REVIEW

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        @current_user = user
+      end
+    end
+  end
+
+  def sign_out
+    session.delete(:user_id)
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+    @current_user = nil
```
Comment (Oghenebrume50) [inline app/helpers/sessions_helper.rb]: You have a setter method already for current_use, why set the variable explicitly again? :thinking:

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class Post < ApplicationRecord
+  belongs_to :user
+  validates :title, presence: true
+  validates :content, presence: true
```
Comment (Oghenebrume50) [inline app/models/post.rb]: While it is good to validate fields in the model class, do not forget to create these constraints in the schema also, currently you don't have any there. Read about Race conditions, a  quick googling would show you alot.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  attr_accessor :token
+  before_create :remember
+  has_many :posts
+
+  validates :username, presence: true, length: { maximum: 20 }
+  validates :email, presence: true, uniqueness: true
```
Comment (Oghenebrume50) [inline app/models/user.rb]: See my comment above on race conditions

**Comment by Redvanisation** [comment]
**_Implemented the user sign up functionality and fixed the issues mentioned in the previous review._**

**Comment by Redvanisation** [comment]
Thank you for the review guys

**Pull request merged**